depends/packages/native_b2.mk

```diff
@@ -0,0 +1,19 @@
+package=native_b2
+$(package)_version=$(boost_version)
+$(package)_download_path=$(boost_download_path)
+$(package)_file_name=$(boost_file_name)
+$(package)_sha256_hash=$(boost_sha256_hash)
+$(package)_build_subdir=tools/build/src/engine
+$(package)_toolset_$(host_os)=gcc
+ifneq (,$(findstring clang,$($(package)_cxx)))
```

Comment:
This doesn't seem to be working as expected. When I pass `CC=clang CXX=clang++`, `native_b2` is still built using the GCC toolset (Boost is built with Clang). I modified this to use a similar change as you've made above, i.e:
```diff
diff --git a/depends/packages/native_b2.mk b/depends/packages/native_b2.mk
index 184c203d9..aaa37cdcf 100644
--- a/depends/packages/native_b2.mk
+++ b/depends/packages/native_b2.mk
@@ -4,9 +4,10 @@ $(package)_download_path=$(boost_download_path)
 $(package)_file_name=$(boost_file_name)
 $(package)_sha256_hash=$(boost_sha256_hash)
 $(package)_build_subdir=tools/build/src/engine
-$(package)_toolset_$(host_os)=gcc
 ifneq (,$(findstring clang,$($(package)_cxx)))
-   $(package)_toolset_$(host_os)=clang
+$(package)_toolset_$(host_os)=clang
+else
+$(package)_toolset_$(host_os)=gcc
 endif
```

However that doesn't seem to work either. 

If I just set `$(package)_toolset_$(host_os)=clang`, then it spits out:
```bash
Building native_b2...
###
###
### Using 'clang' toolset.
###
###
g++ --version
g++ (Debian 10.1.0-6) 10.1.0
```
but given the version number that still looks like it's GCC. I might be missing something.

Reply:
Ah, the reason why setting `CC=clang` and `CXX=clang++` doesn't work is because `CC` and `CXX` only control the `host_{CC,CXX}`, and since `native_b2` is a `native` package, its `$(package)_cxx` derives its value from `build_CXX`. So setting `build_CXX=clang` should work.

As for the case of just setting `$(package)_toolset_$(host_os)=clang`, `$(package)_toolset_$(host_os)` strictly depends on `$(package)_cxx`, so changing that by itself won't influence `$(package)_cxx`. Since it's a strict dependency, we should probably remove this variable at some point in the future and replace it with a simple `$(if $(findstring clang,$($(package)_cxx)),clang,gcc)` to avoid confusion.

Reply:
Yep, thanks for clarifying.